docs/async-await.md

```diff
@@ -0,0 +1,232 @@
+# Proposal: Async-await support
+
+## Introduction
+
+With the introduction of [async/await][SE-0296] in Swift 5.5, it is now possible to write asynchronous code without the need for callbacks. 
+
+Language support for [`AsyncSequence`][SE-0298] also allows for writing functions that return values over time.
+
+We would like to explore how we could offer APIs that make use of these new language features to allow users to run HTTPRequest using these new idioms.
+
+This proposal describes what these APIs could look like and explores some of the potential usability concerns.
+
+## Proposed API additions
+
+### New `HTTPClientRequest` type
+
+The proposed new `HTTPClientRequest` shall be a simple swift structure.
+
+```swift
+struct HTTPClientRequest {
+  /// The requests url.
+  var url: String
+  
+  /// The request's HTTPMethod
+  var method: HTTPMethod
+  
+  /// The request's headers
+  var headers: HTTPHeaders
+
+  /// The request's body
+  var body: Body?
+    
+  init(url: String) {
```

Comment:
Only allowing `url` to be set in the `init` might be a little too restrictive. This pattern makes a lot of sense for large configuration objects etc. where adding a new `init` for each new piece of configuration quickly becomes a burden for maintainers. 

I don't know if that's justified here since this is likely to be used much more frequently so we should make it easy to use even if the maintenance cost is higher. Moreover I suspect there's much less scope for adding new properties to the request.

On the other hand, the "convenience" APIs might cover this just fine.